apps/src/templates/studioHomepages/UiTips.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import UiTip from './UiTip';
+import Dialog from '../Dialog';
+
+const UiTips = React.createClass({
+  propTypes: {
+    userId: React.PropTypes.number,
+    tipId: React.PropTypes.string,
+    showInitialTips: React.PropTypes.bool,
+    beforeDialog: React.PropTypes.object,
+    afterDialog: React.PropTypes.object,
+    tips: React.PropTypes.array
+  },
+
+  getInitialState() {
+    let tipsShowing = [];
+
+    let showInitialTips = this.props.showInitialTips;
+
+    // We might start by showing the "before" dialog.
+    const showingDialog = (showInitialTips && this.props.beforeDialog) ? "before" : null;
+
+    // If we want to show the initla tips but there is no dialog, show the initial tips immediately.
+    if (showInitialTips && !showingDialog) {
+      this.props.tips.map((tip, index) => {
+        if (tip.type === "initial") {
+          tipsShowing[index] = true;
+        }
+      });
+    }
+
+    return {
+      showInitialTips: showInitialTips,
+      tipsShowing: tipsShowing,
+      showingDialog: showingDialog
+    };
+  },
+
+  closeClicked(index) {
+    // Update state so it's no longer showing.
+    let newTipsShowing = [...this.state.tipsShowing];
+    newTipsShowing[index] = false;
+    let newState = {...this.state, tipsShowing: newTipsShowing};
+
+    // Once all the initial tips are dismissed, tell the server to update the user
+    // so it's recorded.
+    if (this.state.showInitialTips) {
+      let tipRemaining = false;
+      newState.tipsShowing.forEach(tipShowing => {
+        if (tipShowing) {
+          tipRemaining = true;
+        }
+      });
+
+      if (!tipRemaining) {
+        $.post(
+          `/api/v1/users/${this.props.userId}/post_ui_tip_dismissed`,
+          { tip: this.props.tipId }
+        );
+
+        // Show a concluding dialog if there is one.
+        if (this.props.afterDialog) {
+          newState = {...newState, showingDialog: "after"};
+        }
+
+        // Don't show initial tips any more.
+        newState = {...newState, showInitialTips: false};
+      }
+    }
+
+    this.setState(newState);
+  },
+
+  afterDialogCancel() {
+    let newState = {...this.state, showingDialog: null};
+    this.setState(newState);
```

Comment:
You don't need to copy the entire state when calling `setState`. `setState` will do the merge for you. So this can just be:

```jsx
this.setState({showingDialog: null});
```